src/main/java/uk/ac/cam/cl/dtg/segue/dao/content/ChoiceOrikaConverter.java

```diff
@@ -72,6 +74,8 @@ public ChoiceDTO convertTo(final Choice source, final Type<ChoiceDTO> destinatio
             return super.mapperFacade.map(source, GraphChoiceDTO.class);
         } else if (source instanceof StringChoice) {
             return super.mapperFacade.map(source, StringChoiceDTO.class);
+        } else if (source instanceof RegexPattern) {
+            return super.mapperFacade.map(source, StringChoiceDTO.class);
```

Comment:
This isn't right. Nothing intentionally needs the DTO object, and so it shouldn't even be needed at all. Unfortunately, to serialise the question for the frontend, we do convert it to a DTO and need all the choices to be DTOs too. These are immediately `@JsonIgnore`-d for all question types except `isaacMultiChoiceQuestion`, but the type hierarchy means that they do have to exist in the DTO just to be immediately discarded.

I _think_, since it will never be used, making this map to the `RegexPatternDTO` is the 'right' thing to do here, since a string choice shouldn't work (but sometimes will because both usually only have the `value` field set and so the mapper does manage to coerce it; if a `RegexPattern` existed with `matchWholeString` set, this would likely explode).
But it is annoying to have to have and use a class that serves absolutely no useful purpose :/

Reply:
This shouldn't affect the validator's ability to expect an inbound `StringChoice` because we trust the user's declared `Choice` subtype, and we use the _question type's_ declared validator which is manually set with the annotation.

Reply:
The whole system is all such a mess . . . 
😞

Reply:
The fact I missed this twice also hints at the complexity! :/

Reply:
Fixed in most recent commit - this was just a typo that I missed.